### ISSUE TITLE
yum_package fails on fedora bento boxes

### ISSUE THREAD

**Issue by artem-sidorenko** [chef/chef#5492]
## Description

If you use fedora 23/34 bento boxes and try to install some package, it will fail
## Chef Version

12.15.19
## Platform Version

Fedora 23 or 24, bento boxes, v.2.2.9
## Replication Case

``` bash
$ chef generate cookbook test
$ cd test
$ echo "package 'screen'" >> recipes/default.rb
```

Update `.kitchen.yml` with following platforms:

``` yml
platforms:
  - name: fedora-23
  - name: fedora-24
```

Try to converge:

``` bash
$ kitchen converge fedora-23
...
       Recipe: test::default
         * yum_package[screen] action install[2016-10-26T21:58:01+00:00] WARN: Yum executable interpreter is /bin/bash. Falling back to default python.


           ================================================================================
           Error executing action `install` on resource 'yum_package[screen]'
           ================================================================================

           Errno::ENOENT
           -------------
           No such file or directory - /usr/bin/python

           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/test/recipes/default.rb

             7: package 'screen'

           Compiled Resource:
           ------------------
           # Declared in /tmp/kitchen/cache/cookbooks/test/recipes/default.rb:7:in `from_file'

           yum_package("screen") do
             package_name "screen"
             action [:install]
             retries 0
             retry_delay 2
             default_guard_interpreter :default
             declared_type :package
             cookbook_name "test"
             recipe_name "default"
             flush_cache {:before=>false, :after=>false}
           end

           Platform:
           ---------
           x86_64-linux


       Running handlers:
       [2016-10-26T21:58:01+00:00] ERROR: Running exception handlers
       Running handlers complete
       [2016-10-26T21:58:01+00:00] ERROR: Exception handlers complete
       Chef Client failed. 0 resources updated in 02 seconds
       [2016-10-26T21:58:01+00:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/chef-stacktrace.out
       [2016-10-26T21:58:01+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
       [2016-10-26T21:58:01+00:00] ERROR: yum_package[screen] (test::default line 7) had an error: Errno::ENOENT: No such file or directory - /usr/bin/python
       [2016-10-26T21:58:01+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
...
```

It looks like the issue is [here](https://github.com/chef/chef/blob/db57131ad383076391b9df32d5e9989cfb312d58/lib/chef/provider/package/yum/yum_cache.rb#L104).  We assume `/usr/bin/python` is there for the execution of `yum_dump`. On fedora 23/24 python2 is not there:

```
[root@default-fedora-23 ~]# ls /usr/bin/python*
/usr/bin/python3  /usr/bin/python3.4  /usr/bin/python3.4m
```

**Comment by artem-sidorenko** [comment]
Still if `yum-dump.py` gets updated with python3 syntax, there are no python yum modules if yum is not installed.

The best way is still the separate dnf provider. In the meantime I would provide a PR to bento to include yum with fedora boxes

**Comment by lamont-granquist** [comment]
dup of #3201?

**Comment by artem-sidorenko** [comment]
@lamont-granquist they are different: #3201 is about dnf provider, this issue is about broken package resource on bento fedora boxes. However the resolution of #3201 would resolve this issue as well

**Comment by lamont-granquist** [comment]
Yeah, so the bento boxes being broken is owned by the bento repo and that issue you cut there is more appropriate.  I don't see how this helps me fix #3201 so I'll close it, if anyone hits this page they can still follow the links to the open issues.

**Comment by artem-sidorenko** [comment]
@lamont-granquist I'm fine with closing it as the clear resolution is #3201 anyway. Just as a note, bento boxes are not broken at all :-) They just make the issue with `yum-dump` visible
